apps/sms/services/js/message_manager.js

```diff
@@ -421,33 +421,41 @@ var MessageManager = {
 
   markMessagesRead: function mm_markMessagesRead(list, isRead = true) {
     if (!this._mozMobileMessage || !list.length) {
-      return;
+      return Promise.reject(new Error('The mozMobileMessage API is not available or list is empty.'));
```

Comment:
i think there won't be any case where list is empty and markMessageRead is called.